service/controller/handler/configversion/create.go

```diff
@@ -0,0 +1,85 @@
+package configversion
+
+import (
+	"context"
+
+	"github.com/ghodss/yaml"
+	"github.com/giantswarm/apiextensions/v3/pkg/annotation"
+	"github.com/giantswarm/microerror"
+
+	controllerkey "github.com/giantswarm/config-controller/service/controller/key"
+)
+
+func (h *Handler) EnsureCreated(ctx context.Context, obj interface{}) error {
+	app, err := controllerkey.ToAppCR(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	value, ok := app.GetAnnotations()[annotation.ConfigVersion]
+	if ok {
+		h.logger.Debugf(ctx, "App CR %#q is already annotated with %#q: %s", app.Name, annotation.ConfigVersion, value)
+		h.logger.Debugf(ctx, "cancelling handler")
+		return nil
+	}
+
+	if app.Spec.Catalog == "" {
+		h.logger.Debugf(ctx, "App CR %#q has no .Spec.Catalog set", app.Name)
+		h.logger.Debugf(ctx, "cancelling handler")
+		return nil
+	}
+
+	if app.Spec.Catalog == "releases" {
+		h.logger.Debugf(ctx, "App CR %#q has a \"releases\" catalog set", app.Name)
```

Comment:
When is the pause annotation removed in that case?

Reply:
I moved annotation removal around a bit. Now it's removed from releases here and from the other resources once the config version is set and config generated.